kits/arm/example_impedance_control_cartesian.cpp

```diff
@@ -0,0 +1,187 @@
+  // Make sure you are running this from the correct directory!
+  params.hrdf_file_ = "kits/arm/hrdf/T-arm.hrdf";
+
+  // Create the Arm Object
+  auto arm = arm::Arm::create(params);
+  while (!arm) {
+    arm = arm::Arm::create(params);
+  }
+
+  // Load the gains file that is approriate to the arm
+  arm -> loadGains("kits/arm/gains/T-arm.xml");
+
+  // Create and configure the ImpedanceController plugin
+
+  // NOTE: Angle wraparound is an unresolved issue which can lead to unstable behaviour for any case involving rotational positional control. 
+  //       Make sure that the rotational gains are high enough to prevent large angular errors (greater than pi/2). The gains provided in these examples are well behaved.
+  //       Interacting with the end-effector in these examples is perfectly safe.
+  //       However, ensure that nothing prevents the wrist's actuators from moving, and DO NOT place your fingers between them. 
+
+  hebi::experimental::arm::PluginConfig impedance_config("ImpedanceController", "ImpedanceController");
+  impedance_config.float_lists_["kp"] = {300.0, 300.0, 300.0, 0.0, 0.0, 0.0};
+  impedance_config.float_lists_["kd"] = {5.0, 5.0, 5.0, 0.0, 0.0, 0.0};
+  impedance_config.float_lists_["ki"] = {20.0, 20.0, 20.0, 0.0, 0.0, 0.0};
+  impedance_config.float_lists_["i_clamp"] = {10.0, 10.0, 10.0, 0.0, 0.0, 0.0}; // Clamp on the end-effector wrench and NOT on the integral error
+  impedance_config.bools_["gains_in_end_effector_frame"] = true;
+
+  auto impedance_plugin = hebi::experimental::arm::plugin::ImpedanceController::create(impedance_config);
+  if (!impedance_plugin) {
+    std::cerr << "Failed to create ImpedanceController plugin." << std::endl;
+    return -1;
+  }
+
+  // Initialize variables used to clear the commanded position and velocity in every cycle
+  hebi::GroupCommand& command = arm->pendingCommand();
+
+  auto num_joints = arm->robotModel().getDoFCount();
+  Eigen::VectorXd pos_nan(num_joints), vel_nan(num_joints);
+  pos_nan.fill(std::numeric_limits<double>::quiet_NaN());
+  vel_nan.fill(std::numeric_limits<double>::quiet_NaN());
+
+  // Add the plugin to the arm
+  if (!arm->addPlugin(std::move(impedance_plugin))) {
+    std::cerr << "Failed to add ImpedanceController plugin to arm." << std::endl;
+    return -1;
+  }
+
+  //////////////////////////
+  //// MobileIO Setup //////
+  //////////////////////////
+
+  // Set up MobileIO
+  std::unique_ptr<util::MobileIO> mobile = util::MobileIO::create(params.families_[0], "mobileIO");
+  if (!mobile)
+  {
+    std::cout << "couldn't find mobile IO device!\n";
+    return 1;
+  }
+  mobile->setButtonMode(1, util::MobileIO::ButtonMode::Momentary);
+  mobile->setButtonLabel(1, "❌");
+  mobile->setButtonMode(2, util::MobileIO::ButtonMode::Toggle);
+  mobile->setButtonLabel(2, "💪");
+
+  std::string instructions;
+  instructions = "                    Cartesian demo";
+  
+  // Clear any garbage on screen
+  mobile->clearText(); 
```

Comment:
This is good for now; two notes for the future:
1. We might want to change to "send with acknowledgement" behavior so we are robust to a dropped packet here.
2. We can use the `sendLayout` mobile IO capabilities in the future, and just have this stored in a file.  (This is in the C API but hasn't been released in C++ yet)